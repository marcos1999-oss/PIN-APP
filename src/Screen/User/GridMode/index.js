import React from "react";
import {
  FlatList,
  RefreshControl,
  View,
  TouchableOpacity,
  Animated,
  Easing,
  Dimensions,
} from "react-native";
import { Item, Input, Content } from "native-base";
import numeral from 'numeral'
import styles from "./style";
import VIPCard from "../../../Components/VIPCard";
import DealCard from "../../../Components/DealCard";
import NoDealsView from "../NoDeals";
//Redux
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  getPostsPagination,
} from '../../../redux/selectors/index';
import * as postActions from '../../../redux/actions/postActions';
import {
  verticalScale,
} from "../../../Utils/scaling";

//GridView

class GridMode extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      searchValue: '',
      refreshing: false,
      isEnded: false,
    };
  }

  onPressDeal = (item) => {
    this.props.navigation.navigate("UserAdView", { isUser: true });
  };

  _keyExtractor = (item, index) => item.id.toString();

  distance = (destLat, destLong) => {
    if (this.props.me !== undefined && destLat !== undefined && destLong !== undefined) {
      const lat1 = this.props.me.latitude;
      const lon1 = this.props.me.longitude;
      const lat2 = destLat;
      const lon2 = destLong;
      const R = 6371; // km (change this constant to get miles)
      const dLat = (lat2 - lat1) * Math.PI / 180;
      const dLon = (lon2 - lon1) * Math.PI / 180;
      const a = Math.sin(dLat / 2) * Math.sin(dLat / 2) +
        Math.cos(lat1 * Math.PI / 180) * Math.cos(lat2 * Math.PI / 180) *
        Math.sin(dLon / 2) * Math.sin(dLon / 2);
      const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
      const d = (R * c) / 1.60934;
      if (d < 1) return Math.round(d.toFixed(2) * 10) / 10;
      else return Math.round(d);
    }
  }

  _renderItem = ({ item, index }) => {
    const distance = this.distance(item.firm.lat, item.firm.lng);
    return (
      <CustomItem
        index={index + 1}
        style={[
          styles.cardContainer,
          (index + 1) % 2 == 0
            ? styles.evenCardContainer
            : styles.oddCardContainer,
        ]}
        onPress={({ item }) => this.onPressDeal({ item })}
      >
        {item.id == 9 ? (
          <VIPCard />
        ) : (
            <DealCard
              image={item.photo.url}
              title={item.title}
              viewsCount={numeral(item.viewsCount).format('0,0')}
              likesCount={numeral(item.likesCount).format('0,0')}
              distance={distance}
            />
          )}
      </CustomItem>
    );
  };

  _onRefresh = () => {
    this.setState({ refreshing: true });
    // console.log("is refreshing");
    setTimeout(() => {
      this.setState({ refreshing: false });
    }, 5000);
  };

  _onChangeText = (searchValue) => {
    this.setState({ searchValue }, () => {
      this.props.onSearch(this.state.searchValue);
    }
    );
  };

  renderEmpty = () => {
    return (<NoDealsView />)
  };

  handlePagination() {
    if (this.props.pagination.currentPage !== this.props.pagination.total) {
      this.props.postActions.fetchPosts({
        currentPage: this.props.pagination.currentPage + 1,
        perPage: this.props.pagination.perPage,
      });
    }
  }

  render() {
    if (this.state.isEnded) {
      return this.renderEmpty()
    } else if (this.props.posts.length === 0 && !this.props.hasSearch) {
      return this.renderEmpty()
    }
    return (
      <View style={styles.container}>
        {this.props.hasSearch && (
          <Item regular style={styles.searchContainer}>
            <Input
              placeholder="Find what you looking forâ€¦"
              style={styles.searchInput}
              value={this.state.searchValue}
              onChangeText={this._onChangeText}
            />
          </Item>
        )}
        <View style={{ flex: 1 }}>
          <FlatList
            data={this.props.posts}
            numColumns={2}
            showsVerticalScrollIndicator={false}
            renderItem={this._renderItem.bind(this)}
            keyExtractor={(item, idx) => item.id}
            contentContainerStyle={styles.dealsContainer}
            onEndReachedThreshold={0.2}
            onEndReached={() => this.handlePagination()}
            getItemLayout={(data, index) => (
              { length: verticalScale(247.67), offset: verticalScale(247.67) * index, index }
            )}
            refreshControl={
              <RefreshControl
                refreshing={this.state.refreshing}
                onRefresh={this._onRefresh.bind(this)}
                title="refreshing ..."
              />
            }
          />
        </View>
      </View>
    );
  }
}

export class CustomItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      easeAnimated: new Animated.Value(0),
    };
  }

  componentDidMount() {
    // this.easeAnimate.setValue(0);
    Animated.timing(this.state.easeAnimated, {
      toValue: 1,
      duration: 1000,
      delay: this.props.index * 500,
      easing: Easing.out(Easing.quad),
      useNativeDriver: true,
    }).start();
  }

  render() {
    const transform = [
      {
        translateY: this.state.easeAnimated.interpolate({
          inputRange: [0, 1],
          outputRange: [styles.cardContainer.height * 2, 0],
        }),
      },
    ];
    return (
      <TouchableOpacity onPress={this.props.onPress}>
        <Animated.View style={[this.props.style, { transform }]}>
          {this.props.children}
        </Animated.View>
      </TouchableOpacity>
    );
  }
}

const mapStateToProps = state => ({
  pagination: getPostsPagination(state),
});

const mapDispatchToProps = dispatch => ({
  postActions: bindActionCreators(postActions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(GridMode)
