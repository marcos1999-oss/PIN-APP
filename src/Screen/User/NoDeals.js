import React from 'react';
import {Platform, View, Text} from "react-native";
import Pulse from "../Business/Home/Pulse";
import {Colors} from "../../Themes";
import {moderateScale, verticalScale} from "../../Utils/scaling";

const NoDealsView = () => {
    const IS_ANDROID = Platform.OS === "android";
    return(
        <View style={{
            flex: 1,
            alignItems: "center",
            flexDirection: "column",
            paddingVertical: verticalScale(30),
        }}>
            <View style={{ flex: 1}}>
                <Pulse
                    color={Colors.background}
                    diameter={400}
                    wrapperStyle={{
                        justifyContent: 'center',
                        alignItems: 'center',
                    }}
                    contentView={
                        <View
                            style={{
                                backgroundColor: Colors.background,
                                width: 100,
                                height: 100,
                                borderRadius: 50,
                                justifyContent: 'center',
                                alignItems: 'center'
                            }}
                        >
                            <Text
                                style={{
                                    fontFamily: IS_ANDROID
                                        ? "SignPainter_HouseScript"
                                        : "SignPainter",
                                    fontSize: moderateScale(28),
                                    color: "white",
                                }}
                            >
                                scopin
                            </Text>
                        </View>
                    }
                />
            </View>
            <View style={{
                height: 100,
                justifyContent: 'center',
                alignItems: 'center'
            }}>
                <Text style={{
                    fontSize: 19,
                    textAlign: 'center',
                    color: "#C4C4C4"
                }}>Currently no new deals{"\n"}pass by locations to see more deals</Text>
            </View>
        </View>
    )
};

export default NoDealsView;
