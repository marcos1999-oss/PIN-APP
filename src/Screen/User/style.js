import React from "react";
import { StyleSheet, Platform } from "react-native";
import { verticalScale, scale, moderateScale } from "../../Utils/scaling";

const IS_IOS = Platform.OS === "ios";
export default StyleSheet.create({
  header: {
    flexDirection: "row",
    paddingHorizontal: scale(15),
    paddingTop: verticalScale(IS_IOS ? 5 : 10),
    justifyContent: "space-between",
  },
  headerIcon: {
    width: verticalScale(35),
    height: verticalScale(35),
  },
  blueDot: {
    position: "absolute",
    bottom: 0,
    left: IS_IOS ? 0 : verticalScale(-11.5),
    width: verticalScale(12),
    height: verticalScale(12),
    borderRadius: verticalScale(6),
    borderWidth: verticalScale(2),
    borderColor: "white",
    backgroundColor: "#44C7F7",
  },
  heartBufferLayer: {
    width: verticalScale(35),
    height: verticalScale(35),
    borderRadius: verticalScale(17.5),
    borderWidth: 0.4,
    borderColor: "#E85858",
    position: "absolute",
    left: 0,
    top: 0,
    backgroundColor: "#E858584C",
  },
  content: {},
  controlPanel: {
    flexDirection: "row",
    alignSelf: "center",
    justifyContent: "space-between",
    marginTop: verticalScale(10),
    // backgroundColor: 'green',
  },
  controlButton: {
    paddingHorizontal: scale(15),
    paddingVertical: verticalScale(10),
  },
  activeControlIcon: {
    fontSize: moderateScale(16),
  },
  inactiveControlIcon: {
    fontSize: moderateScale(16),
    color: "#C4C4C4",
  },
  middleControlButton: {
    paddingHorizontal: scale(20),
    paddingVertical: verticalScale(10),
    borderLeftWidth: scale(0.5),
    borderRightWidth: scale(0.5),
    borderRadius: 0,
    borderColor: "#707070",
  },
});
