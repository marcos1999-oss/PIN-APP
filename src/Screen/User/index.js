import React from "react"
import {
  View,
  SafeAreaView,
  TouchableOpacity, Animated,
  Easing,
  Platform,
  PermissionsAndroid,
} from "react-native"
import { connect } from "react-redux"
import { Button } from "native-base"
import PushNotificationIOS from "@react-native-community/push-notification-ios";
import SwipeMode from "./SwipeMode"
import GridMode from "./GridMode"
const SIZE = responsiveHeight(7);
import { bindActionCreators } from 'redux'
import { map } from 'lodash'
import regionMonitor from '@hkpuits/react-native-region-monitor';
import Geolocation from 'react-native-geolocation-service';
import { setData, getData } from '../../redux/utils/storage';
import { CachedImage } from 'react-native-cached-images';
import { _ } from 'lodash';
import firebase from 'react-native-firebase';

import {
  getMe,
  getUserCoordinates,
  getPosts,
  getSearchedPosts,
  getViewingPost,
  getPostsMeta,
  getNearestPins,
  getPinsMeta,
  getPostsPagination,
} from '../../redux/selectors/index'
import * as authActions from "../../redux/actions/authActions"
import * as postActions from "../../redux/actions/postActions"
import * as pinActions from "../../redux/actions/pinActions"

/*Style*/
import style from "./style.js"
import { verticalScale } from "../../Utils/scaling"
import CustomIcon from "../../CustomIcon"
import { responsiveHeight } from "../../Utils/dimensions"
import AsyncStorage from "@react-native-community/async-storage";


class UserScreen extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      activeControl: 1,
      isClick: false,
      hasLocationPermission: false,
      shownLocationPermission: false,
      pendingNotifications: false,
    };

    this.growAnimated = new Animated.Value(0);
  }

  componentWillMount() {

    this.preload();

    getData('pending_notifications').then((pendingNotifications) => {
      this.setState({ pendingNotifications: pendingNotifications });
    });
  }

  componentDidMount = async () => {

    if (Platform.OS === 'ios') {
      regionMonitor.onRegionChange((event) => {
        const { didEnter, didExit, region, location } = event;
        const { identifier } = region;

        if (didEnter) {
          // const { latitude, longitude } = location;

          Geolocation.getCurrentPosition((position) => {
            const { latitude, longitude } = position.coords;

            this.props.pinActions.visitPin({ id: identifier, latitude, longitude });
          });
        }
      });
    }

    this.getToken();
    await this.requestLocationPermission();
  };

  async getToken() {
    let fcmToken = await AsyncStorage.getItem('fcmToken');
    if (!fcmToken) {
      fcmToken = await firebase.messaging().getToken();
      if (fcmToken) {
        // user has a device token
        await AsyncStorage.setItem('fcmToken', fcmToken);
        this._onRegister(fcmToken);
      }
    }
  }

  _onRegister(token) {

    this.props.authActions.updateAccount({
      userType: 'user',
      params: { push_notification_token: token }
    });
  }

  _onNotification(notification) {
    console.log(`Message: ${notification.getMessage()};\n
      badge: ${notification.getBadgeCount()};\n
      sound: ${notification.getSound()};\n
      category: ${notification.getCategory()};\n
      content-available: ${notification.getContentAvailable()}.`);

    if (Platform.OS === 'ios') {
      PushNotificationIOS.getApplicationIconBadgeNumber((n) => {
        PushNotificationIOS.setApplicationIconBadgeNumber(n + 1);
      });
    }

    this.setState({ pendingNotifications: true }, () => {
      setData('pending_notifications', true);
    });
  }

  preload() {
    if (!this.props.postsMeta.loading && !this.props.postsMeta.loaded) {
      this.props.postActions.fetchPosts({
        currentPage: this.props.pagination.currentPage,
        perPage: this.props.pagination.perPage
      });
    }
  }

  trackLocation() {

    if (this.state.hasLocationPermission) {

      Geolocation.watchPosition(
        (position) => {
          const { latitude, longitude } = position.coords;
          this.props.authActions.locationSuccess({ latitude, longitude });
          this.props.authActions.updateAccount({
            userType: 'user',
            params: { latitude: latitude, longitude: longitude }
          });

          if (!this.props.pinsMeta.loadingNear) {
            this.props.pinActions.fetchNearestPins({
              params: {
                lat: latitude,
                lng: longitude,
              },
              onSuccess: this.setupRegions,
            });
          }
        },
        (error) => {
          console.log(error.code, error.message);
        },
        {
          enableHighAccuracy: true,
          maximumAge: 0,
          distanceFilter: 100,
        }
      );
    }
  }

  requestLocationPermission = async () => {
    try {
      if (Platform.OS === "android" && Platform.Version >= 23) {
        const granted = await PermissionsAndroid.request(
          PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION,
          {
            title: 'Location Permission',
            message: 'Scopin needs access to your location',
            buttonNegative: 'Cancel',
            buttonPositive: 'OK',
          },
        );

        if (granted === PermissionsAndroid.RESULTS.GRANTED) {
          this.setState(
            { hasLocationPermission: true, shownLocationPermission: true },
            this.trackLocation
          );
        } else {
          this.setState({ hasLocationPermission: false, shownLocationPermission: true });
        }
      } else {
        this.setState(
          { hasLocationPermission: true },
          this.trackLocation
        );
      }
    } catch (err) {
      console.warn(err);
    }
  };

  setupRegions = () => {
    const pins = this.props.pins;

    regionMonitor.clearRegions().then(() => {

      map(pins, (pin) => {
        const center = {
          latitude: pin.lat,
          longitude: pin.lng,
        };
        const radius = pin.range * 1609.344;
        const identifier = pin.id;
        regionMonitor.addCircularRegion(center, radius, identifier).then(() => {
        }).catch((error) => {
          console.log(error);
        });
      });
    });
  };

  searchPosts = (query) => {
    this.props.postActions.searchPosts({ params: { params: { q: query } } });
  };

  renderContent = () => {
    switch (this.state.activeControl) {
      case 1:
        return <SwipeMode data={this.props.posts} initialCardIndex={0} {...this.props} />;
      case 2:
        return <GridMode {...this.props} />;
      default:
        return <GridMode hasSearch {...this.props} posts={this.props.searchedPosts} onSearch={this.searchPosts} />;
    }
  };

  pressHeartIcon() {
    this.setState({ isClick: true });

    this.growAnimated = new Animated.Value(0);
    Animated.timing(this.growAnimated, {
      toValue: 1,
      duration: 300,
      useNativeDriver: true,
      easing: Easing.linear,
    }).start();

    setTimeout(() => {
      this.props.navigation.navigate("UserFavorites", { onDidFocus: () => { this.setState({ isClick: false }) } });
    }, 300);

    if (Platform.OS === 'ios') {
      PushNotificationIOS.setApplicationIconBadgeNumber(0);
      this.setState({ pendingNotifications: false }, () => {
        setData('pending_notifications', false);
      });
    }
  }

  render() {
    const { activeControl, pendingNotifications } = this.state;
    const { me } = this.props;
    const changeScale = this.growAnimated.interpolate({
      inputRange: [0, 1],
      outputRange: [1, SIZE],
    });

    if (this.state.isClick) {
      return (
        <View style={{
          flex: 1,
          alignItems: 'flex-end',
          backgroundColor: 'white',
        }}>
          <Animated.View
            style={[{
              height: SIZE,
              width: SIZE,
              marginTop: -SIZE,
              borderRadius: responsiveHeight(10),
            }, { backgroundColor: this.state.isClick ? '#5F92F3' : 'transparent' },
            { transform: [{ scale: changeScale }] }]}
          />
        </View>
      )
    }
    const isProfileIncomplete = _.some(me, _.isEmpty);
    return (
      <View style={{ flex: 1 }}>
        <SafeAreaView>
          <View style={style.header}>
            {
              isProfileIncomplete && (
                <View
                  style={{
                    width: 10,
                    height: 10,
                    backgroundColor: "#F40100",
                    borderRadius: 50,
                    position: "absolute",
                    marginTop: 10,
                    marginLeft: 50,
                    zIndex: 9,
                  }}
                />
              )
            }
            <TouchableOpacity
              // transparent
              style={{
                width: verticalScale(35),
                height: verticalScale(35),
                borderRadius: verticalScale(17.5),
                overflow: "hidden",
                alignItems: "center",
                justifyContent: "center",
              }}
              onPress={() =>
                this.props.navigation.navigate("UserProfile")
              }
            >
              <CachedImage
                source={{ uri: this.props.me.photo.url }}
                style={{
                  // flex: 1,
                  width: verticalScale(37),
                  height: verticalScale(37),
                }}
              />
            </TouchableOpacity>

            <Button
              rounded
              onPress={this.pressHeartIcon.bind(this)}
              style={{
                ...style.headerIcon,
                alignItems: "center",
                justifyContent: "center",
                backgroundColor: "#5F92F3",
              }}
            >
              {/* <Animated.View style={{ ...style.heartBufferLayer, transform }} /> */}
              <CustomIcon
                name="heart3"
                size={verticalScale(13)}
                color="white"
              />
              {pendingNotifications === true ? <View style={style.blueDot} /> : <View />}
            </Button>

          </View>
        </SafeAreaView>
        <View style={style.controlPanel}>
          <Button
            transparent
            style={style.controlButton}
            onPress={() => this.setState({ activeControl: 1 })}
          >
            <CustomIcon
              name="layoutBlock"
              style={
                activeControl == 1
                  ? style.activeControlIcon
                  : style.inactiveControlIcon
              }
            />
          </Button>
          <Button
            transparent
            style={style.middleControlButton}
            onPress={() => this.setState({ activeControl: 2 })}
          >
            <CustomIcon
              name="list"
              style={
                activeControl == 2
                  ? style.activeControlIcon
                  : style.inactiveControlIcon
              }
            />
          </Button>
          <Button
            transparent
            style={style.controlButton}
            onPress={() => this.setState({ activeControl: 3 })}
          >
            <CustomIcon
              name="magnifyingGlass"
              style={
                activeControl == 3
                  ? style.activeControlIcon
                  : style.inactiveControlIcon
              }
            />
          </Button>
        </View>

        {this.renderContent()}
      </View>
    );
  }
}

const mapStateToProps = state => {
  return {
    me: getMe(state),
    userCoordinates: getUserCoordinates(state),
    posts: getPosts(state),
    searchedPosts: getSearchedPosts(state),
    post: getViewingPost(state),
    postsMeta: getPostsMeta(state),
    pins: getNearestPins(state),
    pinsMeta: getPinsMeta(state),
    pagination: getPostsPagination(state),
  }
};

const mapDispatchToProps = dispatch => {
  return {
    authActions: bindActionCreators(authActions, dispatch),
    postActions: bindActionCreators(postActions, dispatch),
    pinActions: bindActionCreators(pinActions, dispatch),
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(UserScreen);
