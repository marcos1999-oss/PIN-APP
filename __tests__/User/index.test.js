/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import 'react-native';
import React from 'react';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import mockComponent from 'react-native/jest/mockComponent';

import User from '../../src/Screen/User';

const mockStore = configureMockStore();

jest.mock('react-native-cached-images', () => {
  return {
    CustomCachedImage: mockComponent('Image'),
    CachedImage: mockComponent('Image'),
    DocumentDir: () => {},
    ImageCache: {
      get: {
        clear: () => {}
      }
    }
  }
});

jest.mock('@react-native-community/push-notification-ios', () => {
  return {
    addEventListener: jest.fn(),
    requestPermissions: jest.fn(),
    getApplicationIconBadgeNumber: jest.fn(),
    setApplicationIconBadgeNumber: jest.fn() 
  }
})

jest.mock('@hkpuits/react-native-region-monitor', () => {
  return {
    onRegionChange: jest.fn(),
    clearRegions: jest.fn(),
    addCircularRegion: jest.fn()
  }
})

jest.mock('react-native-geolocation-service', () => {
  return {
    getCurrentPosition: jest.fn(),
    watchPosition: jest.fn(),
  }
})

test('renders correctly', () => {
  const store = mockStore({
    auth: {
      me: {
        photo: {
          url: 'https://www.easydrawingtips.com/wp-content/uploads/2019/10/spiky_male_hair_drawing.jpg',
        }
      }
    },
    posts: {
      meta: {
        loading: true
      },
      data: {},
    },
    offers: {
      data: {},
    },
    firms: {
      data: {},
    },
    features: {
      data: {},
    },
    holidays: {
      data: {},
    },
    schedules: {
      data: {},
    },
    pins: {
      near: {},
    }
  });
  const tree = renderer.create(
      <Provider store={store}>
        <User />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
