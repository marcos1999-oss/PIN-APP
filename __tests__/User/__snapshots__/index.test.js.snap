// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 30,
          "paddingTop": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 35,
            "height": 70,
            "justifyContent": "center",
            "opacity": 1,
            "overflow": "hidden",
            "width": 70,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://www.easydrawingtips.com/wp-content/uploads/2019/10/spiky_male_hair_drawing.jpg",
            }
          }
          style={
            Object {
              "height": 74,
              "width": 74,
            }
          }
        />
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#5F92F3",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 57,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 2,
            "flexDirection": "row",
            "height": 70,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "width": 70,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 26,
              },
              undefined,
              Object {
                "fontFamily": "PinAppIcon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <View />
      </View>
    </View>
  </RCTSafeAreaView>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 20,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingHorizontal": 30,
          "paddingTop": 6,
          "paddingVertical": 20,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "fontSize": 24,
            },
            Object {
              "fontFamily": "PinAppIcon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#707070",
          "borderLeftWidth": 1,
          "borderRadius": 0,
          "borderRightWidth": 1,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingHorizontal": 40,
          "paddingTop": 6,
          "paddingVertical": 20,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#C4C4C4",
              "fontSize": 24,
            },
            Object {
              "fontFamily": "PinAppIcon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingHorizontal": 30,
          "paddingTop": 6,
          "paddingVertical": 20,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#C4C4C4",
              "fontSize": 24,
            },
            Object {
              "fontFamily": "PinAppIcon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "paddingVertical": 60,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
            },
            Object {
              "alignItems": "center",
              "bottom": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "top": 0,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 400,
                "width": 400,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#5F92F3",
                "borderRadius": 50,
                "height": 100,
                "justifyContent": "center",
                "width": 100,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "SignPainter",
                  "fontSize": 42,
                }
              }
            >
              scopin
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 100,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#C4C4C4",
            "fontSize": 19,
            "textAlign": "center",
          }
        }
      >
        Currently no new deals
        

        pass by locations to see more deals
      </Text>
    </View>
  </View>
</View>
`;
